bugsnag-android-core/src/main/java/com/bugsnag/android/internal/MemoryMappedOutputStream.kt

```diff
@@ -0,0 +1,61 @@
+package com.bugsnag.android.internal
+
+import java.io.File
+import java.io.OutputStream
+import java.io.RandomAccessFile
+import java.nio.MappedByteBuffer
+import java.nio.channels.FileChannel
+
+/**
+ * OutputStream wrapper around a shared memory file.
+ *
+ * Memory-mapped files are maintained by the operating system, and so they are guaranteed to match
+ * the last data written to the memory region, even if the app crashes.
+ *
+ * The memory mapped file is fixed to the length specified at instantiation, meaning that
+ * write() will throw java.nio.BufferOverflowException if the file is too full.
+ */
+class MemoryMappedOutputStream(
+    file: File,
+    private val bufferSize: Int,
+    private val clearedByteValue: Byte
+    ) : OutputStream() {
+
+    private val raf: RandomAccessFile = RandomAccessFile(file, "rw")
```

Comment:
The Javadoc for this class states that the rwd/rws mode ensures all updates are made synchronously to the storage device, and specifically mentions a system crash as a use-case. Is this something we'd want to consider or is rw sufficient?

Reply:
I think it's better to not have synchronous, since OS crashes are extremely rare, and synchronous writes take a lot of time.